### PR TITLE
Feature/expose plugin version in config file

### PR DESCRIPTION
**Pull request by dikhan**
## Proposed changes

Please add as many details as possible about the change here. Does this Pull Request resolve any open issue? 
If so, please make sure to link to that issue:
                                                              
Fixes: #124 

## Type of change

What type of change does your code introduce to the provider? Please put an `x` (w/o heading/trailing white spaces) 
in the boxes that apply:

- [x] Bug-fix (change that fixes current functionality)
- [x] New feature (change that adds new functionality)

## Checklist

Please put an `x` (w/o heading/trailing white spaces) in the boxes that apply:

- [x] I have read and followed the [CONTRIBUTING](https://github.com/dikhan/terraform-provider-api/blob/master/.github/CONTRIBUTING.md) guidelines
- [x] I have made sure code compiles correctly
- [x] I have run 'make test' locally from the terraform_provider_api folder and no errors were found
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added/updated necessary documentation (if appropriate)

### CODE REVIEW

File: openapi/plugin_config_services.go [hunk 1]
```diff
@@ -52,3 +69,20 @@ func (s *ServiceConfigV1) GetSchemaPropertyConfiguration(schemaPropertyName stri
 	}
 	return nil
 }
+
+// Validate makes sure the configuration is valid:
+// - if the user has specified an OpenAPI plugin version, and if the plugin does not match the version then something is off
+func (s *ServiceConfigV1) Validate(runningPluginVersion string) error {
+	if !govalidator.IsURL(s.SwaggerURL) {
```
Comment (simar7) [inline openapi/plugin_config_services.go]: what if swaggerURL is empty

File: openapi/plugin_config_services.go [hunk 2]
```diff
@@ -52,3 +69,20 @@ func (s *ServiceConfigV1) GetSchemaPropertyConfiguration(schemaPropertyName stri
...
+		// fall back to try to load the swagger file from disk in case the path provided is a path to a file on disk
+		if _, err := os.Stat(s.SwaggerURL); os.IsNotExist(err) {
+			return fmt.Errorf("service swagger URL configuration not valid ('%s'). URL must be either a valid formed URL or a path to an existing swagger file stored in the disk", s.SwaggerURL)
+		}
+	}
+	if s.PluginVersion != "" {
```
Comment (simar7) [inline openapi/plugin_config_services.go]: remove - should just check directly to runningPluginVersion

File: openapi/plugin_config_services_test.go [hunk 3]
```diff
@@ -46,3 +64,67 @@ func TestPluginConfigSchemaV1IsSecureSkipVerifyEnabled(t *testing.T) {
...
+			err := serviceConfiguration.Validate(runningPluginVersion)
+			Convey("Then the error returned should be nil", func() {
+				So(err, ShouldBeNil)
+			})
+		})
+	})
+	Convey("Given a ServiceConfigV1 containing an invalid swagger URL pointing at a file store in the disk", t, func() {
+		var serviceConfiguration ServiceConfiguration
+		swaggerFile, _ := ioutil.TempFile("", "")
+		defer os.RemoveAll(swaggerFile.Name()) // clean up
```
Comment (simar7) [inline openapi/plugin_config_services_test.go]: missing file handle closure - best to defer the call to remove and swaggerFile.Close() https://golang.org/pkg/io/ioutil/#example_TempFile

File: docs/plugin_configuration_schema.md [hunk 4]
```diff
@@ -69,7 +69,8 @@ Describes the configurations available on a single service.
 
 Field Name | Type | Description
 ---|:---:|---
-swagger-url | `string` | **Required.** Defines the location where the swagger document is hosted. The value must be either a valid formatted URL or a path to a swagger file stored in the disk 
+swagger-url | `string` | **Required.** Defines the location where the swagger document is hosted. The value must be either a valid formatted URL or a path to a swagger file stored in the disk
```
Comment (simar7) [inline docs/plugin_configuration_schema.md]: s/stored in the disk/stored on disk

File: docs/plugin_configuration_schema.md [hunk 5]
```diff
@@ -69,7 +69,8 @@ Describes the configurations available on a single service.
...
+plugin_version | `string` | Defines the plugin version. If this value is specified, the openapi plugin version executed must match this value; otherwise and error will be thrown at runtime.
```
Comment (simar7) [inline docs/plugin_configuration_schema.md]: s/otherwise and error/otherwise an error

File: openapi/plugin_config_schema.go [hunk 6]
```diff
@@ -53,14 +52,6 @@ func (p *PluginConfigSchemaV1) Validate() error {
 	if p.Version != "1" {
 		return fmt.Errorf("provider configuration version not matching current implementation, please use version '1' of provider configuration specification")
 	}
-	for k, v := range p.Services {
```
Comment (simar7) [inline openapi/plugin_config_schema.go]: documenting for future reference: this is a bug fix unrelated to this PR

File: openapi/plugin_config_services.go (hunk 1 above)
Comment (simar7) [reply openapi/plugin_config_services.go]: handled

File: openapi/plugin_config_schema.go (hunk 6 above)
Comment (dikhan) [reply openapi/plugin_config_schema.go]: updated the issue description to cover this too

**Pull request merged**